bundles/io/org.eclipse.smarthome.io.transport.mqtt.test/src/test/java/org/eclipse/smarthome/io/transport/mqtt/test/MqttBrokerConnectionTests.java

```
@@ -0,0 +1,185 @@
+    }
+
+    @Test
+    public void testConstructor() {
+        // Test tcp and ssl URLs
+        try {
+            MqttBrokerConnection a = new MqttBrokerConnection("name", "tcp://123.123.123.123", false);
+            MqttBrokerConnection b = new MqttBrokerConnection("name", "ssl://123.123.123.123", true);
+            assertFalse(a.isTextualConfiguredBroker());
+            assertTrue(b.isTextualConfiguredBroker());
+        } catch (ConfigurationException e) {
+            fail("Unexpected configuration exception: " + e.getMessage());
+        }
+        // Test an invalid URL
+        try {
+            @SuppressWarnings("unused")
+            MqttBrokerConnection ignored = new MqttBrokerConnection("name", "unsupported://123.123.123.123", false);
+            fail("Configuration exception expected: No valid url protocol");
+        } catch (ConfigurationException e) {
+        }
+        // Test an invalid name
+        try {
+            @SuppressWarnings("unused")
+            MqttBrokerConnection ignored = new MqttBrokerConnection(" ", "tcp://123.123.123.123", false);
+            fail("Configuration exception expected: No valid name");
+        } catch (ConfigurationException e) {
+        }
+    }
+
+    @Test
+    public void messageConsumerTests() {
+        final String url = "tcp://123.123.123.123";
+        final String name = "TestName12@!";
+        MqttBrokerConnection a;
+        try {
+            a = new MqttBrokerConnection(name, url, false);
+        } catch (ConfigurationException e) {
+            fail();
+            return;
+        }
+        assertFalse(a.isConsumers());
+        MqttMessageSubscriber subscriber = new MqttMessageSubscriber() {
+            @Override
+            public void processMessage(String topic, byte[] payload) {
+
+            }
+
+            @Override
+            public String getTopic() {
+                return "topic";
+
+            }
+        };
+        try {
+            a.addConsumer(subscriber);
```

same as in line 71/72. Just call `addConsumer` and let the test fail due to the anyhow thrown exception.